Fix/tab item alignment

RubenSmn opened the pull request:
# Description

Improves the item alignment in a tab.

Changes made:

- Vertical align items in a tab, introduced `display: flex`
- Fix misplaced hover state on buttons
- Fix permission buttons spacing
- Buttons now have a consistent height / width

|Before|
|:-:|
|![_home_ruben_Development_forks_min_index html](https://user-images.githubusercontent.com/31519867/233373804-3ccf34fa-fff4-4d1d-ad27-b38d644e6765.png)|
|After|
|![_home_ruben_Development_forks_min_index html](https://user-images.githubusercontent.com/31519867/233373462-3a4805b6-c30d-4b86-beac-cacd5a417bbd.png)

*I forced some active states to show the icons.*

PalmerAL commented:
Thanks for the PR! It's odd things are so misaligned in your original screenshot - those buttons are mostly centered on macOS, and I recall them looking much better than that on Windows and Linux also. Switching to flexbox is a good solution though.

Two bugs I've noticed so far:

The hover effect on the tab close button is broken:

<img width="104" alt="Screenshot 2023-04-25 at 11 58 50 PM" src="https://user-images.githubusercontent.com/10314059/234473943-4bad3722-89ba-4431-94e1-17412f8874df.png">

There's something weird going on with the layout of tabs with permission icons when selected:


https://user-images.githubusercontent.com/10314059/234474109-6e8cded2-62e4-41df-ac2c-d1f409db8fb2.mov

RubenSmn commented:
Hey thanks for the feedback! I'll get that fixed. 

Might be worth mentioning I'm using Linux: ZorinOS

RubenSmn commented:
@PalmerAL as you might be able to tell from the screenshots the permission icon changed a bit, its not fully round anymore. Should it still be round?

RubenSmn commented:
I've fixed the issues @PalmerAL.

- used `gap` for consistent spacing between items inside a `tab-item`
- title has 50% more spacing and does not 'touch' the right side
- used dynamic width for the back/forward button container as it was occupying unnecessary space

Let me know if there are anymore improvements I can make!

PalmerAL commented:
Sorry, two more things:

1. The spacing seems broken when the tab is wider than the title and there are icons:

<img width="1470" alt="Screenshot 2023-05-12 at 9 13 02 PM" src="https://github.com/minbrowser/min/assets/10314059/8d9adf12-01f2-4152-9173-270740605a4e">

Removing the `justify-content: space-between` on the tab item sort of fixes this, but then the close button is in the wrong place:
<img width="1469" alt="Screenshot 2023-05-12 at 9 18 11 PM" src="https://github.com/minbrowser/min/assets/10314059/938b522c-efd1-4235-bb18-961e82e7eae7">


2. Previously, the forward button would appear with an animation - how difficult would it be to add this back?


https://github.com/minbrowser/min/assets/10314059/f0b3208c-2199-4c2a-baca-55473614be22

RubenSmn commented:
No need for any apology, thank you for finding these edge cases. I found 2 others myself

- spacing between tab close button and the the unsecure webpage indicator
- permissions indicator was not responsible for more than 1 permission e.g. camera and microphone

I've fixed the title spacing and the placement of the close button. Changed the fixed width of the navigation buttons to a more appropriate size (was occupying unnecessary space) which brings back the transition / animation.

PalmerAL commented:
This looks really good! I only see one minor issue left: the spacing on the sides of the tab item is a bit uneven, because the `gap` applies to the `tab-icon-area` even when no icons are showing:

<img width="245" alt="Screenshot 2023-05-16 at 9 42 19 PM" src="https://github.com/minbrowser/min/assets/10314059/bf6945b3-001b-4623-aef2-8a0b7398c706">

This means an extra character or two of the tab title is hidden when it doesn't need to be. It's not a huge issue though, if you don't see a simple way of fixing it (I don't at the moment) I'm fine leaving it as-is.

RubenSmn commented:
Great! I was going to solve this by checking if the `tab-icon-area` had no children and then apply  the gap width as negative margin. But I found that it always has a `i` element as a secure indicator, so now wer'e checking if the area only contains the `i` element and then applying the style.

PalmerAL commented:
I don't think that works, because the icon area also contains the tab close button, but what do you think of the change in 1b679548bd5e3bd1b2136b843e2b98e3bd44ba57 ?

RubenSmn commented:
Ah okay, this works better but I'm having a issue with hovering. Might be something just on linux as I'm not seeing the close icon on your tabs and I assume they probably appear when hovering a tab.

[Kooha-2023-05-18-11:58:23.webm](https://github.com/minbrowser/min/assets/31519867/40d7602b-a8b5-4c01-9958-fd2fde0d40fd)

RubenSmn commented on css/tabBar.css:
```diff
@@ -153,14 +159,23 @@ body.linux #menu-button {
+  gap: 0.5rem;
+  order: 1;
+}
+.tab-item:not(:hover) .tab-icon-area:has(.tab-close-button:only-child) {
```
I came up with this

```suggestion
.linux .tab-item .tab-icon-area:has(.tab-close-button:only-child),
:not(.linux)
  .tab-item:not(:hover)
  .tab-icon-area:has(.tab-close-button:only-child) {
```

RubenSmn commented on css/tabBar.css:
```diff
@@ -305,12 +323,9 @@ body.windows.dark-theme .navbar-action-button:not(:disabled):hover {
   font-size: 1.25em;
-  padding: 0.3em 0.45em;
-  margin-top: 0.15em;
+  box-sizing: border-box;
```
Should the close icon not be the icon most to the right side?

![image](https://github.com/minbrowser/min/assets/31519867/b4fa9061-e719-4c66-b861-0c042685153e)

```suggestion
  box-sizing: border-box;
  order: 1;
```

PalmerAL replied:
Yes it should.

RubenSmn commented:
Yes I have a touch screen, the rule you've provided does the trick @PalmerAL!

I couldn't find anymore issues, let me know if you need any adjustments.

PalmerAL commented:
It all looks good to me! Thanks for all the revisions on this!
